pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -173,6 +188,35 @@ func (i *CommandInitOption) Validate(parentCommand string) error {
 	return nil
 }
 
+func (i *CommandInitOption) validateExternalEtcd(_ string) error {
+	if (i.ExternalEtcdClientCertPath == "" && i.ExternalEtcdClientKeyPath != "") ||
+		(i.ExternalEtcdClientCertPath != "" && i.ExternalEtcdClientKeyPath == "") {
+		return fmt.Errorf("etcd client cert and key should be specified both or none")
+	}
+	return nil
+}
+
+func (i *CommandInitOption) isExternalEtcdProvided() bool {
+	return i.ExternalEtcdServers != ""
+}
+
+// Validate Check that there are enough flags to run the command.
+//
+//nolint:gocyclo
+func (i *CommandInitOption) Validate(parentCommand string) error {
+	if i.KarmadaAPIServerAdvertiseAddress != "" {
+		if netutils.ParseIPSloppy(i.KarmadaAPIServerAdvertiseAddress) == nil {
+			return fmt.Errorf("karmada apiserver advertise address is not valid")
+		}
+	}
+
+	if i.isExternalEtcdProvided() {
+		return i.validateExternalEtcd(parentCommand)
+	} else {
+		return i.validateBundledEtcd(parentCommand)
```

Comment:
I am not sure whether `Bundled` is the most appropriate adjective.
`External` looks easy to associate with `Internal`.
Let us for others opinion.
Otherwise LGTM.
@chaosi-zju  @RainbowMango

Reply:
Since karmadactl does essentially the same thing as kubeadm, I usually compare karmadactl with kubeadm, which is divided into `etcd.local` and `etcd.external` in kubeadm.

 `etcd.local` and `etcd.external` also using by Karmada operator, so i would recommend using `Local` instead of `Bundled`.

Reply:
Agree with @liangyuanpeng. 

So, we just need to change the function name, right?

Reply:
bundled -> local
done